app/api/web/fetch_file.coffee

```diff
@@ -0,0 +1,96 @@
+#include all the required dependencies
+util = require('../../common/util')
+appconfig = require('../../config/appconfig').appconfig
+fs = require('fs')
+request = require('request')
+csrf = require('csurf')
+async = require('async')
+prettysize = require('prettysize')
+path = require('path')
+tmpDir = appconfig.getTemporaryDirectory()
+fileLimits = appconfig.getFileLimits()
+filestores = {}
+mimeRegex = /.(avi|wmv|flv|mpg|3gp|mkv|mp4|mpeg|mpeg-1|mpeg-2|mpeg-3|mpeg-4|mp3|wav|xlsx?|zip|7z|docx?|pptx?|pdf)$/i
+
+exports.init = (app) ->
+	
+	app.post '/fetch/', csrf({ cookie:true }), (req,res) ->
+		#check request came
+		console.log "Request recieved"
+		#set file name from req.body
+		fileName = path.basename(req.body.url)
```

Comment:
`basename(..)` always extracts the _last part of the url_, so unless the url is **null** , the fileName won't be null.